spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/sql/TestIcebergVersionFunction.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark.sql;
+
+import org.apache.iceberg.IcebergBuild;
+import org.apache.iceberg.spark.SparkCatalogConfig;
+import org.apache.iceberg.spark.SparkTestBaseWithCatalog;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class TestIcebergVersionFunction extends SparkTestBaseWithCatalog {
+
+  @Parameterized.Parameters(name = "catalogConfig = {0}")
+  public static Object[][] parameters() {
+    return new Object[][] {{SparkCatalogConfig.SPARK}};
+  }
+
+  private final String fullyQualifiedSystemNamespace;
+
+  public TestIcebergVersionFunction(SparkCatalogConfig catalogConfig) {
+    super(catalogConfig);
+    this.fullyQualifiedSystemNamespace = catalogConfig.catalogName() + ".system";
+  }
+
+  @Test
+  public void testIcebergVersionFunctionReturnsCorrectResult() {
+    // Check that value correctly loads from file, even in codegen etc
+    String buildVersion = IcebergBuild.version();
+    Assert.assertNotEquals(
+        "iceberg_version should not return undefined",
+        "undefined",
+        scalarSql("SELECT %s.iceberg_version()", fullyQualifiedSystemNamespace));
+
+    Assert.assertEquals(
+        "Should be able to use the Iceberg version function from the fully qualified system namespace",
+        buildVersion,
+        scalarSql("SELECT %s.iceberg_version()", fullyQualifiedSystemNamespace));
```

Comment:
Why not `SELECT %s.system.iceberg_version()`? I don't see a need to have a separate variable for the fully-qualified namespace.

Reply:
Updated.